### PR TITLE
Implement Float64 support for some DSP functions 

### PR DESCRIPTION
**Pull request by jbr-smtg**
Hello all,

The following pull request is similar to https://github.com/ARM-software/CMSIS_5/pull/1307 but contains the following fixes:
- Branch Rebased
- Fix new files version number

Don't hesitate to let me know everything that needs to be changed/modified.

Thanks in advance for your time!

### CODE REVIEW

**Comment by grasci-arm** [comment]
Can one of the admins verify this patch?

File: CMSIS/DSP/Source/BasicMathFunctions/arm_dot_prod_f64.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+void arm_dot_prod_f64(
+  const float64_t * pSrcA,
+  const float64_t * pSrcB,
+        uint32_t blockSize,
+        float64_t * result)
+{
+  uint32_t blkCnt;                               /* Loop counter */
+  float64_t sum = 0.0f;                          /* Temporary return variable */
```
Comment (christophe0606) [inline CMSIS/DSP/Source/BasicMathFunctions/arm_dot_prod_f64.c:69]: The f suffix must be removed. All constants must be double in f64 code.
There are probably a lot of other files with similar problems.

File: CMSIS/DSP/Source/FilteringFunctions/arm_fir_f64.c [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+ * limitations under the License.
+ */
+
+#include "dsp/filtering_functions.h"
+
+/**
+  @ingroup groupFilters
+ */
+
+/**
```
Comment (christophe0606) [inline CMSIS/DSP/Source/FilteringFunctions/arm_fir_f64.c:35]: This should be removed. The doxygen group is defined already by the f32 function.
I think there may be similar problems with other functions.

Generally the doxygen group is introduced by the f32 function and should not be duplicated in the f64.

**Review by christophe0606 — CHANGES_REQUESTED** [review]
In the previous (closed) pull request there were some cmake updates.
I no more see them.

The new files should be added to the cmake if the cmake is not using a file(GLOB SRCF64 "./*_f64.c")

File: CMSIS/DSP/Source/BasicMathFunctions/arm_dot_prod_f64.c (hunk 1 above)
Comment (jbr-smtg) [reply CMSIS/DSP/Source/BasicMathFunctions/arm_dot_prod_f64.c]: Thanks! I just fixed it.

File: CMSIS/DSP/Source/FilteringFunctions/arm_fir_f64.c (hunk 2 above)
Comment (jbr-smtg) [reply CMSIS/DSP/Source/FilteringFunctions/arm_fir_f64.c:35]: I removed the docygen group.

**Comment by christophe0606** [comment]
@jbr-smtg But you have the problem of float constant and doxygen group in much more than the files I have highlighted.

For the doxygen defgroup I think you may have the problem in nearly all the f64 files if they come from a copy of the f32 file.

For the float issue, each f32 file containing a constant will give the issue.

For instance, arm_mean_f64.c has the issue (0.0f constant to init the sum).

I have not reviewed all the files for those problems.

File: CMSIS/DSP/Source/BasicMathFunctions/arm_dot_prod_f64.c (hunk 1 above)
Comment (jbr-smtg) [reply CMSIS/DSP/Source/BasicMathFunctions/arm_dot_prod_f64.c]: I did a new commit that should have fixed most of them.

**Comment by christophe0606** [comment]
@jbr-smtg Can you add the new files to the CMakeLists.txt (except when the CMakeLists.txt are using a pattern like file(GLOB SRCF64 "./*_f64.c") in BasicMaths).

Because I need to try to rebuild with a few different compilers and targeting several different cores before I can accept the pull request. On older compilers we often have some surprises requiring small changes to the code for building.

I'll accept the pull request only when I can check it is building on all our configurations and that the documentation generated from the Doxygen is correct.

Thank you.

**Comment by jbr-smtg** [comment]
Thank-you :)

**Comment by christophe0606** [comment]
@jbr-smtg I have found some additional issues:

**arm_max_no_idx_f64**

`float64_t   maxValue = F32_MIN`

It should be F64_MIN.

**arm_chebyshev_distance_f64**

There is a remaining 0.0f

After those changes, I should be able to merge the pull request and start working on the tests.

**Comment by christophe0606** [comment]
Merged.

Now I am going to work on the tests.

**Comment by jbr-smtg** [comment]
Perfect :)
Thanks again!

**Pull request merged**